terraform/provider/pkg/provider/provider.go

```diff
@@ -147,6 +141,22 @@ func requestAccessToken(scope, audience, signedToken string) (string, error) {
 	return accessTokenResp.AccessToken, nil
 }
 
+type TfAwsCredentialsProvider struct {
+	appCreds *stscreds.AssumeRoleProvider
+	ctx      context.Context
```

Comment:
I think its an antipattern to pass a context as part of a struct attribute. The typical convention is to always include the context as part of the first argument of the function, like you do in getAWsCredsProvider

https://groups.google.com/g/golang-nuts/c/xRbzq8yzKWI?pli=1

Reply:
👍 I'll update `GetCredentials` to take a context arg then it wont be necessary as a struct field